OnvifSharp/Discovery/WSDiscovery.cs

```diff
@@ -83,35 +125,59 @@ XmlProbeReponse DeserializeResponse (string xml)
 			}
 		}
 
-		IEnumerable<DiscoveryDevice> CreateDevices (XmlProbeReponse response, IPEndPoint remoteEndpoint)
+		DiscoveryDevice CreateDevice (XmlProbeReponse response, IPEndPoint remoteEndpoint)
 		{
-			foreach (var probeMatch in response.Body.ProbeMatches) {
-				var discoveryDevice = new DiscoveryDevice ();
-				discoveryDevice.Address = remoteEndpoint.Address;
-				discoveryDevice.XAdresses = ConvertToList (probeMatch.XAddrs);
-				discoveryDevice.Types = ConvertToList (probeMatch.Types);
-				discoveryDevice.Model = ParseModelFromScopes (probeMatch.Scopes);
-				discoveryDevice.Name = ParseNameFromScopes (probeMatch.Scopes);
-				yield return discoveryDevice;
-			}
+			var discoveryDevice = new DiscoveryDevice ();
+			string scopes = response.Body.ProbeMatches[0].Scopes;
+			discoveryDevice.Address = remoteEndpoint.Address.ToString ();
+			discoveryDevice.Model = Regex.Match (scopes, "(?<=hardware/).*?(?= )").Value;
+			discoveryDevice.Mac = GetMacAddress (discoveryDevice.Address);
+			discoveryDevice.Mfr = ParseMfrFromScopes (scopes);
+			string xaddr = response.Body.ProbeMatches[0].XAddrs;
+			Uri uri = new Uri (xaddr);
+			discoveryDevice.Port = uri.Port;
+			return discoveryDevice;
 		}
 
-		IEnumerable<string> ConvertToList (string spacedListString)
+		string GetMacAddress (string ipAddress)
 		{
-			var strings = spacedListString.Split (null);
-			foreach (var str in strings) {
-				yield return str.Trim ();
+			string macAddress = String.Empty;
+			System.Diagnostics.Process pProcess = new System.Diagnostics.Process ();
+			pProcess.StartInfo.FileName = "arp";
+			pProcess.StartInfo.Arguments = "-a " + ipAddress;
```

Comment:
I will check that arp command is also available in linux and mac, since this library is designed to be cross-platform. If not I will remove the Mac from the model, unit further investigation.